.eslintrc.js

```diff
@@ -52,6 +73,50 @@ module.exports = {
         }
       }
     },
+    {
+      // Extensions not required for CommonJS require()
+      files: ['**/*.{cjs,js}'],
+      rules: {
+        'import/extensions': [
+          'error',
+          'always',
+          {
+            ignorePackages: true,
+            pattern: {
+              cjs: 'never',
+              js: 'never',
+              mjs: 'never'
+            }
+          }
+        ]
+      }
+    },
```

Comment:
Ahh but I've kept the defaults on purpose

**CommonJS** = [Automatic directory/file/index resolution](https://nodejs.org/api/modules.html#all-together)
**ESM** = [Mandatory file extensions](https://nodejs.org/api/esm.html#mandatory-file-extensions)

Would be fab to make all this into a `@alphagov/eslint-config` package one day? 😍 

For example, path autocompletion in Visual Studio code automatically leaves off the extension (no configs needed) so you'd get red squiggles from the suggested approach here

I'm more than happy to remove the CommonJS rules so mixed extensions or no extensions are fine?

Maybe just keep the ESM ones where we literally need [mandatory file extensions](https://nodejs.org/api/esm.html#mandatory-file-extensions)

Reply:
Just a reminder too that if you set ESLint as your editor/IDE "formatter" it'll auto run `--fix` on save 😮

Reply:
Fair point on adding burden by making enforcing extensions on both, didn't think of that. I think I'd prefer shipping this without it and discuss if we want any kind of rule on the CJS imports: CJS is technically more flexible there but the commit fixing the issues seemed to highlight a good number of places where there was an extension, so this may have been a concious decision (even if just for consistency).

Reply:
Fab, that's done 👍 

Kept the CommonJS fixes in (no extensions) but only the ESM rules remain